src/core/interaction-creator.ts

```diff
@@ -0,0 +1,194 @@
+        };
+    }
+
+    private getRequestBodyForApiFunction(apiFunctionNode: tsMorph.Node) {
+        const requestBodyElement =
+            getParameterWithJsDocFromFunction(apiFunctionNode, 'pact-request-body') ||
+            getVariableWithJsDocFromFunction(apiFunctionNode, 'pact-request-body');
+        if (requestBodyElement) {
+            const requestBodyElementType = requestBodyElement.getType();
+            const basicTypeRepresentationOfRequestBody = getBasicRepresentationOfType(requestBodyElementType, this.sourceFile);
+            return {
+                body: changeObjectRepresentationIntoExample(basicTypeRepresentationOfRequestBody),
+                matchingRules: mapMatchingRepresentation(
+                    changeObjectRepresentationIntoMatchingRules(basicTypeRepresentationOfRequestBody, '$.body'),
+                    '$.body',
+                ),
+            };
+        }
+    }
+
+    private getQueryRequestForApiFunction(apiFunctionNode: tsMorph.Node) {
+        const queryElement =
+            getParameterWithJsDocFromFunction(apiFunctionNode, 'pact-query') ||
+            getVariableWithJsDocFromFunction(apiFunctionNode, 'pact-query');
+        if (queryElement) {
+            const queryElementType = queryElement.getType();
+            const basicTypeRepresentationOfRequestBody = getBasicRepresentationOfType(queryElementType, this.sourceFile);
+            const exampleRepresentationOfQueryObject = changeObjectRepresentationIntoExample(basicTypeRepresentationOfRequestBody);
+            return {
+                query: qs.stringify(exampleRepresentationOfQueryObject),
+                matchingRules: mapMatchingRepresentation(
+                    changeObjectRepresentationIntoMatchingRules(basicTypeRepresentationOfRequestBody, '$.query'),
+                    '$.query',
+                ),
+            };
+        }
+    }
+}
+
+const getFunctionNode = (node: tsMorph.Node) => {
+    switch (node.getKind()) {
+        case ts.SyntaxKind.FunctionDeclaration:
+        case ts.SyntaxKind.MethodDeclaration:
+            return node;
+        case ts.SyntaxKind.VariableDeclaration:
+            const variableDeclarationNode = node
+                .getFirstChildByKind(ts.SyntaxKind.VariableDeclarationList)
+                ?.getFirstChildByKind(ts.SyntaxKind.VariableDeclaration);
+            if (variableDeclarationNode) {
+                return (
+                    variableDeclarationNode.getFirstChildByKind(ts.SyntaxKind.FunctionDeclaration) ||
+                    variableDeclarationNode.getFirstChildByKind(ts.SyntaxKind.ArrowFunction)
+                );
+            }
+            break;
+        case ts.SyntaxKind.PropertyAssignment:
+        case ts.SyntaxKind.PropertyDeclaration:
+            return node.getFirstChildByKind(ts.SyntaxKind.FunctionExpression) || node.getFirstChildByKind(ts.SyntaxKind.ArrowFunction);
+    }
+};
+
+const getResponseTypeFromFunctionBody = (bodyOfFunction: tsMorph.Block) => {
```

Comment:
add return type - here and in another functions

Reply:
OK, added